src/features/common/Modal.tsx

```diff
@@ -17,6 +18,9 @@ export const Modal = ({
   children?: React.ReactNode;
 }) => {
   const modalRef = useRef<HTMLDialogElement>(null);
+  const { isTxLoading, txStatus } = useStakingModalStore();
```

Comment:
⚠️ Can't do this - this is a generic modal that is used in multiple places and should not have staking specific code within it.

The modal should accept a prop, probably something like `isClosable`, (defaulting to `true`). 
You can then set `isClosable={!isTxStatusLoading}` where you're rendering `Modal` in the parent componet.

Reply:
Fixed